### ISSUE TITLE
[Feature Request] Add a H264 backend other than ih264d

### ISSUE THREAD

**Issue by donmor** [cemu-project/Cemu#935]
Platform: Arch Linux on loongarch64

I failed to build Cemu on my platform and found the dependency `ih264d` not been ported to platforms other than x86 and arm yet. I'm not sure if there is any plan to port Cemu to other architectures, and if so, then another backend for H264 decoding is required.

**Comment by Exzap** [comment]
This is more or less a duplicate of #217
ih264d itself is platform independent for the most part. The assumption of x86/ARM is just baked into it because we don't have plans to target any other architecture.

**Comment by Daltz333** [comment]
Adding onto this, it seems that ih264d depends on intel intrinsics that is not implemented on Windows ARM64. This means this issue is a critical blocker for any native Cemu build running on Windows ARM64. 

For reference, this is the error I get using the following cmake flags.

```
cmake -S . -B build -DCMAKE_BUILD_TYPE=release -G Ninja -DCV_ENABLE_INTRINSICS=OFF -DENABLE_SSE=OFF -DENABLE_SSE2=OFF -DENABLE_SSE3=OFF -DENABLE_SSE41=OFF -DENABLE_SSE42=OFF -DENABLE_SSSE3=OFF
```

```
C:\Users\daltz\Documents\Git_Repos\Cemu>cmake --build build
[6/529] Building C object dependencies\ih264d\CMakeFiles\ih264d.dir\common\ih264_chroma_intra_pred_filters.c.obj
FAILED: dependencies/ih264d/CMakeFiles/ih264d.dir/common/ih264_chroma_intra_pred_filters.c.obj
C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\MSVC\1443~1.348\bin\HOSTAR~1\arm64\cl.exe  /nologo -DEMULATOR_HASH=06233e3 -DEMULATOR_VERSION_MAJOR=0 -DEMULATOR_VERSION_MINOR=0 -DEMULATOR_VERSION_PATCH=0 -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DHAS_DIRECTINPUT -DHAS_HIDAPI -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\Vulkan-Headers\include -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\decoder -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common\x86 -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\decoder\x86 /DWIN32 /D_WINDOWS /O2 /Ob2 /DNDEBUG -MT /GL /EHsc /fp:precise /GT /Oi /Ot /GS- /showIncludes /Fodependencies\ih264d\CMakeFiles\ih264d.dir\common\ih264_chroma_intra_pred_filters.c.obj /Fddependencies\ih264d\CMakeFiles\ih264d.dir\ih264d.pdb /FS -c C:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common\ih264_chroma_intra_pred_filters.c
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\immintrin.h(15): fatal error C1189: #error:  This header is specific to X86, X64, ARM64, and ARM64EC targets
[8/529] Building CXX object dependencies\discord-rpc\src\CMakeFiles\discord-rpc.dir\connection_win.cpp.obj
C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\winbase.h(7810): warning C5039: 'TpSetCallbackCleanupGroup': pointer or reference to potentially throwing function passed to 'extern "C"' function under -EHc. Undefined behavior may occur if this function throws an exception.
[11/529] Building CXX object dependencies\discord-rpc\src\CMakeFiles\discord-rpc.dir\discord_register_win.cpp.obj
C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\winbase.h(7810): warning C5039: 'TpSetCallbackCleanupGroup': pointer or reference to potentially throwing function passed to 'extern "C"' function under -EHc. Undefined behavior may occur if this function throws an exception.
[12/529] Building C object dependencies\ih264d\CMakeFiles\ih264d.dir\common\ih264_deblk_edge_filters.c.obj
FAILED: dependencies/ih264d/CMakeFiles/ih264d.dir/common/ih264_deblk_edge_filters.c.obj
C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\MSVC\1443~1.348\bin\HOSTAR~1\arm64\cl.exe  /nologo -DEMULATOR_HASH=06233e3 -DEMULATOR_VERSION_MAJOR=0 -DEMULATOR_VERSION_MINOR=0 -DEMULATOR_VERSION_PATCH=0 -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DHAS_DIRECTINPUT -DHAS_HIDAPI -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\Vulkan-Headers\include -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\decoder -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common\x86 -IC:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\decoder\x86 /DWIN32 /D_WINDOWS /O2 /Ob2 /DNDEBUG -MT /GL /EHsc /fp:precise /GT /Oi /Ot /GS- /showIncludes /Fodependencies\ih264d\CMakeFiles\ih264d.dir\common\ih264_deblk_edge_filters.c.obj /Fddependencies\ih264d\CMakeFiles\ih264d.dir\ih264d.pdb /FS -c C:\Users\daltz\Documents\Git_Repos\Cemu\dependencies\ih264d\common\ih264_deblk_edge_filters.c
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\immintrin.h(15): fatal error C1189: #error:  This header is specific to X86, X64, ARM64, and ARM64EC targets
[19/529] Building CXX object dependencies\cubeb\CMakeFiles\cubeb.dir\src\cubeb_wasapi.cpp.obj
ninja: build stopped: subcommand failed.
```

**Comment by sylveon** [comment]
One should be able to use `USE_SOFT_INTRINSICS` to get software implementations of those intrinsics.

**Comment by Exzap** [comment]
ih264d has an aarch64 implementation but Cemu hardcodes the x86 path at the moment. It's addressed by #1255
